test/cpp/util/channelz_sampler.cc

```diff
@@ -0,0 +1,289 @@
+  return subchannel.ref().subchannel_id();
+}
+int64_t GetSocketID(grpc::channelz::v1::Socket& socket) {
+  return socket.ref().socket_id();
+}
+
+void GetChannelRPC(int64_t channel_id,
+                   grpc::channelz::v1::Channelz::Stub* channelz_stub,
+                   std::queue<grpc::channelz::v1::Channel>& channel_queue) {
+  GetChannelRequest get_channel_request;
+  get_channel_request.set_channel_id(channel_id);
+  GetChannelResponse get_channel_response;
+  ClientContext get_channel_context;
+  channelz_stub->GetChannel(&get_channel_context, get_channel_request,
+                            &get_channel_response);
+  channel_queue.push(get_channel_response.channel());
+}
+
+void GetSubchannelRPC(
+    int64_t subchannel_id, grpc::channelz::v1::Channelz::Stub* channelz_stub,
+    std::queue<grpc::channelz::v1::Subchannel>& subchannel_queue) {
+  GetSubchannelRequest get_subchannel_request;
+  get_subchannel_request.set_subchannel_id(subchannel_id);
+  GetSubchannelResponse get_subchannel_response;
+  ClientContext get_subchannel_context;
+  channelz_stub->GetSubchannel(&get_subchannel_context, get_subchannel_request,
+                               &get_subchannel_response);
+  subchannel_queue.push(get_subchannel_response.subchannel());
+}
+
+void GetChannelDescedence(
+    grpc::channelz::v1::Channelz::Stub* channelz_stub,
+    grpc::channelz::v1::Channel& channel,
+    std::queue<grpc::channelz::v1::Channel>& channel_queue,
+    std::queue<grpc::channelz::v1::Subchannel>& subchannel_queue) {
+  std::cout << "    Channel " << GetChannelID(channel) << " descendence - ";
+  if (channel.channel_ref_size() > 0) {
+    std::cout << "channel: ";
+    for (auto& i : channel.channel_ref()) {
+      int64_t ch_id = i.channel_id();
+      std::cout << ch_id << " ";
+      GetChannelRPC(ch_id, channelz_stub, channel_queue);
+    }
+  }
+  if (channel.subchannel_ref_size() > 0) {
+    std::cout << "subchannel: ";
+    for (auto& i : channel.subchannel_ref()) {
+      int64_t subch_id = i.subchannel_id();
+      std::cout << subch_id << " ";
+      GetSubchannelRPC(subch_id, channelz_stub, subchannel_queue);
+    }
+  }
+  if (channel.socket_ref_size() > 0) {
+    std::cout << "socket: ";
+    for (auto& i : channel.socket_ref()) {
+      int64_t so_id = i.socket_id();
+      std::cout << so_id << " ";
+    }
+  }
+  std::cout << std::endl;
+}
+
+void GetSubchannelDescedence(
+    grpc::channelz::v1::Channelz::Stub* channelz_stub,
+    grpc::channelz::v1::Subchannel& subchannel,
+    std::queue<grpc::channelz::v1::Channel>& channel_queue,
+    std::queue<grpc::channelz::v1::Subchannel>& subchannel_queue) {
+  std::cout << "    Subchannel " << GetSubchannelID(subchannel)
+            << " descendence - ";
+  if (subchannel.channel_ref_size() > 0) {
+    std::cout << "channel: ";
+    for (auto& i : subchannel.channel_ref()) {
+      int64_t ch_id = i.channel_id();
+      std::cout << ch_id << " ";
```

Comment:
similar to above, it would be more interesting to dump the full proto messages, rather than just the channel/subchannel/socket id's

Reply:
Keep the current output format for quickly showing the tree structure, and print out entity (i.e., channel, subchannel, server, socket) data in the end.